Shopify/ruby-lsp
# Add super support for definition and hover

### Motivation

Now that the node context tracks the surrounding method name and we have ancestor linearization, we can provide hover and go to definition for `super` calls.

### Implementation

I split by commit to make reviewing easier:

1. Infer receiver types for `super` calls. Which is the same implementation as instance variables
2. Allow resolving only inherited methods in the index API. When jumping to a super call, we want to resolve the first inherited declaration, ignoring the declaration inside the current class itself
3. Support for definition
4. Support for hover

### Automated Tests

Added tests.

### Manual Tests

1. Launch the LSP on this branch
2. Go to `entry.rb` (lots of super calls for `initialize` there)
3. Verify you can hover and jump to definition on `super`
